main.tf

```diff
@@ -106,133 +231,342 @@ data "aws_iam_policy_document" "secrets_access" {
-  task_execution_policies = length(aws_iam_policy.secrets_access) > 0 ? [aws_iam_policy.secrets_access[0].arn] : []
-  blue_green_deployment_config = {
-    termination_wait_time_after_deployment_success = null // defaults to 15
-    prod_traffic_listener_arns                     = [module.alb.listeners[443].arn]
-    test_traffic_listener_arns                     = []
-    //Note: The `lookup` is used because there have been cases where it can't find the map value when trying to destroy
-    //and that caused the destroy to fail
-    blue_target_group_name  = lookup(module.alb.target_groups, "blue", null) != null ? module.alb.target_groups["blue"].name : null
-    green_target_group_name = lookup(module.alb.target_groups, "green", null) != null ? module.alb.target_groups["green"].name : null
-    service_role_arn        = module.acs.power_builder_role.arn
-  }
-
-  module_depends_on             = [module.alb.alb]
-  role_permissions_boundary_arn = module.acs.role_permissions_boundary.arn
-  log_retention_in_days         = var.log_retention_in_days
-  health_check_grace_period     = var.health_check_grace_period
-  tags                          = local.tags
-}
-
-module "autoscaling" {
-  source             = "github.com/byu-oit/terraform-aws-app-autoscaling?ref=v1.0.1"
-  app_name           = var.app_name
+    actions = ["sts:AssumeRole"]
+  }
+}
+resource "aws_iam_role" "codedeploy_role" {
+  name                 = "${var.app_name}-codedeploy-role"
+  permissions_boundary = var.role_permissions_boundary_arn
+  assume_role_policy   = data.aws_iam_policy_document.codedeploy_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "AWSCodeDeployRole" {
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  role       = aws_iam_role.codedeploy_role.name
+}
+
+resource "aws_codedeploy_deployment_group" "deploymentgroup" {
+  app_name               = aws_codedeploy_app.app.name
+  deployment_group_name  = "${var.app_name}-deployment-group"
+  service_role_arn       = aws_iam_role.codedeploy_role.arn
+  deployment_config_name = "CodeDeployDefault.ECSAllAtOnce"
+
+  ecs_service {
+    cluster_name = aws_ecs_cluster.cluster.name
+    service_name = aws_ecs_service.service.name
+  }
+
+  blue_green_deployment_config {
+    deployment_ready_option {
+      action_on_timeout = "CONTINUE_DEPLOYMENT"
+    }
+    terminate_blue_instances_on_deployment_success {
+      action                           = "TERMINATE"
+      termination_wait_time_in_minutes = 15
+    }
+  }
+
+  deployment_style {
+    deployment_option = "WITH_TRAFFIC_CONTROL"
+    deployment_type   = "BLUE_GREEN"
+  }
+  auto_rollback_configuration {
+    enabled = true
+    events  = ["DEPLOYMENT_FAILURE"]
+  }
+
+  load_balancer_info {
+    target_group_pair_info {
+      prod_traffic_route {
+        listener_arns = [aws_alb_listener.https.arn]
+      }
+      //      TODO test listener traffic
```

Comment:
Should I be worried about this?

Reply:
No. Adding a test listener is easy. That's probably the next enhancement we need to look at.

Reply:
No. Adding a test listener is easy. That's probably the next enhancement we should look at.